kedro/extras/datasets/avro/avro_dataset.py

```diff
@@ -0,0 +1,212 @@
+        >>>             {"name": "col1", "type": "int"},
+        >>>             {"name": "col2", "type": "int"},
+        >>>             {"name": "col3", "type": "int"},
+        >>>         ],
+        >>>     },
+        >>> }
+        >>>
+        >>> # data_set = AVRODataSet(filepath="gcs://bucket/test.avro", schema=schema)
+        >>> data_set = AVRODataSet(filepath="test.csv", save_args=schema)
+        >>> data_set.save(data)
+        >>> reloaded = data_set.load()
+        >>> assert data == reloaded
+
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        schema: Optional[Dict[str, Any]] = None,
+        codec: Optional[str] = "null",
+        version: Version = None,
+        credentials: Dict[str, Any] = None,
+        fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``AVRODataSet`` pointing to a concrete AVRO file
+        on a specific filesystem.
+
+        Args:
+            filepath: Filepath in POSIX format to AVRO file prefixed with a protocol like `s3://`.
+                If prefix is not provided, `file` protocol (local filesystem) will be used.
+                The prefix should be any protocol supported by ``fsspec``.
+                Note: `http(s)` doesn't support versioning.
+            schema: AVRO schema.
+                See details about AVRO schema here:
+                https://avro.apache.org/docs/current/spec.html#schemas
+            codec: AVRO writing codec.
+                See details about AVRO codes here:
+                https://avro.apache.org/docs/current/spec.html#Required+Codecs
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class constructor
+                (e.g. `{"project": "my-project"}` for ``GCSFileSystem``), as well as
+                to pass to the filesystem's `open` method through nested keys
+                `open_args_load` and `open_args_save`.
+                Here you can find all available arguments for `open`:
+                https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem.open
+                All defaults are preserved, except `mode`, which is set to `rb` when loading
+                and to `wb` when saving.
+        """
+        _credentials = deepcopy(credentials) or {}
+        _fs_args = deepcopy(fs_args) or {}
+
+        self._fs_open_args_load = _fs_args.pop("open_args_load", {})
+        self._fs_open_args_load.setdefault("mode", "rb")
```

Comment:
`rb` is already the default value, so no need to set it here.

Reply:
I'd rather leave it as an explicitly specified attribute, what do you think?

Reply:
In all the other datasets we don't explicitly define the load mode if it's using the default, so I'd prefer to keep it consistent and remove it here as well.